cloud/awsprovider.go

```diff
@@ -137,22 +152,46 @@ func (aws *AWS) KubeAttrConversion(location, instanceType, operatingSystem strin
 	return region + "," + instanceType + "," + operatingSystem
 }
 
+type awsKey struct {
+	SpotLabelName  string
+	SpotLabelValue string
+	Labels         map[string]string
+	ProviderID     string
+}
+
+func (k *awsKey) ID() string {
+	parsedProviderID := strings.Split(k.ProviderID, "/")
```

Comment:
If you are expecting it to be of the for `aws:///foo/bar` and you want `bar`, then I would recommend codifying that assumption through `regexp.FindStringSubmatch`; `Split` will not recognize if the providerID switches to `aws://us-east-2a/instances/i-cafebabe` but regexp will

Reply:
Done